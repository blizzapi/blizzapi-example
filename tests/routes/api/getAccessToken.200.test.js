const { BlizzAPI } = require('blizzapi');
const request = require('supertest');
const app = require('../../../src/app');

jest.mock('blizzapi', () => ({
  BlizzAPI: jest.fn().mockImplementation(() => ({
    getAccessToken: () => Promise.resolve('sample access token'),
  })),
}));

describe('/getAccessToken/:region', () => {
  const route = '/api/getAccessToken/us';

  test('should return 200', async () => {
    const response = await request(app).get(route);
    expect(response.statusCode).toBe(200);
  });

  test('should match snapshot', async () => {
    const response = await request(app).get(route);
    expect(response.body).toMatchSnapshot();
  });
});