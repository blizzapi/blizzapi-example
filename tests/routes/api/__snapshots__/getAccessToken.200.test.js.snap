// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/getAccessToken/:region should match snapshot 1`] = `"sample access token"`;
